### ISSUE TITLE
Rewards page (earned by me) shows no data when using pagination

### ISSUE THREAD

**Issue by gordsport** [vegaprotocol/frontend-monorepo#5871]
## Description

Rewards page (earned by me) shows no data when using pagination

## Steps to Reproduce

1. Go to https://governance.vega.xyz/rewards
2. Click on "earned by me"
3. Scroll down to pagination and click to next page
4. See data is not shown in the table

## Expected behavior

I see my data for each epoch

Additionally I am not expecting to see the columns for rewards other than Staking and Infra fees since these are the only rewards relevant for stakers. 

This was covered [here](https://github.com/vegaprotocol/frontend-monorepo/issues/5677) but it looks like we're just showing zeroes instead of removing the columns entirely which is imho misleading because someone might have rewards there if they are trading too. 

## Screenshots

**Page 1:**

<img width="980" alt="Screenshot 2024-02-27 at 08 35 38" src="https://github.com/vegaprotocol/frontend-monorepo/assets/83510148/a75862c1-d2e5-4a77-9c9d-6132a4b021d0">

**When using pagination (Page 2):**

<img width="980" alt="Screenshot 2024-02-27 at 08 36 00" src="https://github.com/vegaprotocol/frontend-monorepo/assets/83510148/a102d0da-f0a7-420b-b1bc-e69a2e82343a">

## Device details

**Desktop (please complete the following information):**

- OS: MacOS 14.2.1
- Browser: Brave 
- Version v1.63.161

**Comment by edd** [comment]
I've unassigned myself as I am not having a lot of success and find the code kind of hard to follow. I believe the problem is here-ish:

https://github.com/vegaprotocol/frontend-monorepo/blob/2e07ada966d5eca6766a061944377ea21532ec31/apps/governance/src/routes/rewards/epoch-individual-rewards/epoch-individual-rewards.tsx#L73-L89

I *think* when the refetch is happening, it is fetching rewards for the previous 'page' worth of epoch rewards, so when it munges them together (https://github.com/vegaprotocol/frontend-monorepo/blob/2e07ada966d5eca6766a061944377ea21532ec31/apps/governance/src/routes/rewards/epoch-individual-rewards/epoch-individual-rewards.tsx#L62-L71) it appears to have no rewards. I *guess* memoization, or the order in which things are triggered, are to blame. 

`acc55c7205cfcd5480e0235acab56a01487a39dc858a641fc04df6ba016870ee` is an example party that should have rewards on page two of epochs.
